File: Java/src/main/java/QuestionBank/Solution601_700/Solution617.java
package QuestionBank.Solution601_700;

import tool.TreeNode;

/**
 * @author windows
 */
public class Solution617 {

    public TreeNode mergeTrees(TreeNode root1, TreeNode root2) {
        if (root1 == null && root2 == null){
            return null;
        }
        if (root1 != null && root2 != null){
            root1.val = root1.val + root2.val;
            if (root1.left != null && root2.left != null) {
                mergeTrees(root1.left, root2.left);
            }
            if (root1.left == null && root2.left != null){
                root1.left = root2.left;
            }
            if (root1.right != null && root2.right != null) {
                mergeTrees(root1.right, root2.right);
            }
            if (root1.right == null && root2.right != null){
                root1.right = root2.right;
            }
        }
        if (root1 == null && root2 != null){
            return root2;
        }
        return root1;
    }

    public static void main(String[] args) {
        Solution617 solution617 = new Solution617();
    }
}

File: C++/src/Solution601_700/Solution617.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution617.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

//Definition for a binary tree node.
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

TreeNode* mergeTrees(TreeNode* root1, TreeNode* root2) {
  if (root1 == nullptr && root2 == nullptr) return nullptr;
  if (root1 == nullptr) return root2;
  if (root2 == nullptr) return root1;
  root1->val += root2->val;
  TreeNode* left = mergeTrees(root1->left, root2->left);
  TreeNode* right = mergeTrees(root1->right, root2->right);
  root1->left = left;
  root1->right = right;
  return root1;
}